ts/features/bonus/bpd/components/bpdCardComponent/BpdCardComponent.tsx

```diff
@@ -111,46 +118,92 @@ type BadgeDefinition = {
 export const BpdCardComponent: React.FunctionComponent<Props> = (
   props: Props
 ) => {
-  const formatStatusBadge = (): BadgeDefinition => {
-    const { period } = props;
+  const { period, totalAmount } = props;
 
-    switch (period.status) {
-      case "Active":
-        return {
-          style: {
-            backgroundColor: IOColors.white
-          },
-          label: I18n.t("bonus.bpd.details.card.status.active")
-        };
-      case "Closed":
-        return {
-          style: {
-            backgroundColor: IOColors.black
-          },
-          label: I18n.t("bonus.bpd.details.card.status.closed")
-        };
-      case "Inactive":
-        return {
-          style: {
-            backgroundColor: IOColors.aqua
-          },
-          label: I18n.t("bonus.bpd.details.card.status.inactive")
-        };
-      default:
-        return {
-          style: {
-            backgroundColor: IOColors.white
-          },
-          label: "-"
-        };
-    }
+  const INITIAL_BADGE_DEFINITION = {
+    style: {
+      backgroundColor: IOColors.white
+    },
+    label: "-"
   };
+  const [amount, setAmount] = React.useState(["0", "00"]);
+  const [statusBadge, setStatusBadge] = React.useState<BadgeDefinition>(
+    INITIAL_BADGE_DEFINITION
+  );
+  const [showLock, setShowLock] = React.useState(false);
+  const [isInGracePeriod, setIsInGracePeriod] = React.useState(false);
```

Comment:
There is a problem with this `state` approach. When the props received will be updated (eg. refresh of data) the component doesn't render again because of the use of `useEffect(..., []);` that will calculate the effect only when the component is mounted (render only the props state when mounted).

In this case the component rendering always depends from the props and maybe we can avoid to use the state.
One alternative could be create some function that calculate the required state, using the props.

For example one alternative could be:
```typescript

type GraphicalState = {
amount: string;
isInGracePeriod: boolean;
showLock: boolean;
statusBadge: BadgeDefinition;
}

const calculateGraphicalState = (props: Props): GraphicalState => {
   switch(props.period.status):
        case "Closed":
            return calculateClosedPeriod(props); // specific code for close period, will return GraphicalState
        case "Current":
            return calculateCurrentPeriod(props); // specific code for current period, will return GraphicalState
        case "Inactive":
            return calculateInactivePeriod(props); // specific code for inactive period, will return GraphicalState
}

export const BpdCardComponent: React.FunctionComponent<Props> = (
  props: Props
) => {
 const {amount, isInGracePeriod, showLock, statusBadge } = calculateGraphicalState(props);

 [... rendering and using the calculated variables ...]
};

```

Reply:
Yup, sounds great! I'm going to rework it in a similar way